src/libxtp/huffmantree.cc

```diff
@@ -0,0 +1,119 @@
+#include <vector>
+#include <list>
+#include <stdlib.h>
+#include <queue>
+#include <votca/xtp/huffmantree.h>
+#include <votca/xtp/glink.h>
+using namespace std;
+
+
+
+namespace votca {namespace xtp {
+
+huffmanTree::huffmanTree()
+{
+}
+
+//https://en.wikipedia.org/wiki/Huffman_coding
+void huffmanTree::makeTree(){
+    if (!events) throw runtime_error("Error in Huffmantree::makeTree : Pointer to Events not set!");
+
+    //queue of the nodes, sorted by probability
+    auto compare = [](huffmanNode * n1, huffmanNode * n2)
+    { return n1->probability>n2->probability;};
+    priority_queue<huffmanNode *,vector<huffmanNode *>, decltype(compare)> queue(compare);
+
+    htree=vector<huffmanNode>(events->size()%2?events->size():events->size()-1);
+
+    auto comp2 = [](GLink * e1, GLink * e2){
+        return e1->rate>e2->rate;
+    };
+    priority_queue<GLink *, vector<GLink *>, decltype(comp2)> eventQueue(comp2);
+
+    int i=0;
```

Comment:
can you remove this counter, and use some other iteration, if not give it a more explicit name